File: contrib/libatscc2py3/ATS2-0.3.2/CATS/reference_cats.py
######
#
# HX-2014-08:
# for Python code translated from ATS
#
######

######
# beg of [reference_cats.py]
######

############################################

def ats2pypre_ref(x): return [x]
def ats2pypre_ref_make_elt(x): return [x]

############################################
#
def ats2pypre_ref_get_elt(ref): return ref[0]
def ats2pypre_ref_set_elt(ref, x0): ref[0] = x0; return
#
def ats2pypre_ref_exch_elt(ref, x0): x1 = ref[0]; ref[0] = x0; return x1
#
############################################

###### end of [reference_cats.py] ######

File: contrib/libatscc2clj/ATS2-0.3.2/CATS/reference_cats.clj
;;
;;;;;;
;
; HX-2016-07:
; for Clojure code
; translated from ATS
;
;;;;;;
;;

;;
;;;;;;
; beg of [reference_cats.clj]
;;;;;;
;;

;; ****** ****** ;;
;;
(defmacro
 ats2cljpre_ref[x] `(atom ~x)
)
(defmacro
 ats2cljpre_ref_make_elt[x] `(atom ~x)
)
;;
;; ****** ****** ;;
;;
(defmacro
 ats2cljpre_ref_get_elt[r] `(deref ~r)
)
(defmacro
 ats2cljpre_ref_set_elt[r x0] `(reset! ~r ~x0)
)
;;
;; ****** ****** ;;

;; end of [reference_cats.clj] ;;
